platform/platform.cpp

```diff
@@ -87,6 +87,11 @@ bool Platform::RmDirRecursively(string const & dirName)
   return res;
 }
 
+void Platform::SetSettingsDirForTesting(string const & path)
+{
+  m_settingsDir = my::AddSlashIfNeeded(path);
```

Comment:
Why do we need this? All methods that use settings dir must use JoinFoldersToPath, which joins path components with a slash.